### ISSUE TITLE
Multi pronged discovery for Everywhere

### ISSUE THREAD

**Issue by tarunsingh5** [KhalisFoundation/sttm-desktop#44]
I think users should have a single discover option which will async discover in multiple ways and populate the list similar to a mac finding wifi. (and sort by distance where short range discovers are up top).

The below are in order of implementation

1) wifi/geo based discovery
2) bluetooth
3) sound

We can also consider a snapchat/qr type approach in the future once the above are completed.

**Comment by inderpreetsingh** [comment]
I am not sure I understand what this is about. @tarunsingh5  @maneetpaul

**Comment by ManjotS** [comment]
@inderpreetsingh This is part of the sync process. 

This issue first requires  desktop to desktop sync (I believe theres a branch  for this, we did it at hackathon to web).

The next nice to have is a remote  admin sync code (which I can add to API), so that sync can effectively be  reversed (ie the  joiner controls the server, vs the server controlling  the  joiner)

After  this, we then want "codeless" discovery.. ie, join a session near me.  This would  use the methods  outlined in OP.

**Comment by tarunsingh5** [comment]
@maneetpaul can we close this ticket since we are going w/ the 6 letter code for now?

**Comment by ManjotS** [comment]
We still want all the other options. The code was just needed to implement the advanced ones.

**Comment by sukhpalhub** [comment]
Agree, there should be a way to use login method to remove navigation when the person navigating is not connected to same network but has login access.

Also can the overlay work similar because sometime the local network has got ip address changing and can't open the overlay as it require access to main device. This happens when there is different devices used for streaming and shabad display.
